examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+                scratch_data.non_matching_fe_values.get_surface_fe_values();
+            const std::vector<double> &JxW_surface =
+              surface_fe_values->get_JxW_values();
+
+            // The accumulation of the surface integrals, including the forcing,
+            // is performed here.
+            for (unsigned int q : surface_fe_values->quadrature_point_indices())
+              {
+                const Tensor<1, dim> &normal =
+                  surface_fe_values->normal_vector(q);
+
+                for (const unsigned int i : surface_fe_values->dof_indices())
+                  {
+                    copy_data.cell_rhs(i) +=
+                      right_hand_side.value(
+                        surface_fe_values->quadrature_point(q)) *
+                      surface_fe_values->shape_value(i, q) * JxW_surface[q];
+
+                    for (const unsigned int j :
+                         surface_fe_values->dof_indices())
+                      {
+                        copy_data.cell_matrix(i, j) +=
+                          1.0 *
+                          (surface_fe_values->shape_value(i, q) *
+                           surface_fe_values->shape_value(j, q)) *
+                          JxW_surface[q];
+                        copy_data.cell_matrix(i, j) +=
+                          (1.0 *
+                           (surface_fe_values->shape_grad(i, q) -
+                            (normal * surface_fe_values->shape_grad(i, q)) *
+                              normal) *
+                           (surface_fe_values->shape_grad(j, q) -
+                            (normal * surface_fe_values->shape_grad(j, q)) *
+                              normal)) *
+                          JxW_surface[q];
+                      }
+                  }
+              }
+
+            // The normal-gradient volume stabilization form needs a bulk cell
+            // integration while other types of stabilization may need face
+            // quadratures, for example. So we check it first.
+            // The cell was provided by the solution's DoF-handler,
+            //  so we recast it as a level set's DoF-handler cell.
+            //  However, it is the same geometric entity of the common
+            //  triangulation.
+            // Next, the copier worker distributes the local contributions from
+            // the CopyData taking into account the constraints. Finally, the
+            // MeshWorker goes over all cells provided by the solutions'
+            // DoF-handler. Note that this includes non-intersected cells as
+            // well,
+            //  but the cell worker does nothing on them.
+            if (stabilization_scheme.needs_cell_worker())
+              {
+                typename DoFHandler<dim>::active_cell_iterator level_set_cell(
+                  &(triangulation),
+                  cell->level(),
+                  cell->index(),
+                  &level_set_dof_handler);
```

Comment:
This might be easier to read if you wrote it as follows:
```suggestion
                typename DoFHandler<dim>::active_cell_iterator level_set_cell
                = typename Triangulation<dim>::active_cell_iterator(cell)
                       .as_dof_handler_iterator(level_set_dof_handler);
```